### PR TITLE
Metric

### LINKED ISSUE #2: Implement Stockfish evals
## Description

* :star2: Goal: Being able to compute stockfish evals (and others) for a game.

## Acceptance Criterion

* [x] The code is documented
* [x] Utilities and class test are written
* [ ] The code was reviewed

## Tasks

* [x] Implement stockfish evals in a `metric` module
* [x] Make the implementation a class similar to `sklearn` metrics
* [x] Add tests to ensure metric correctness

### PR DESCRIPTION
**Pull request by sej2020**
Closes #2

### CODE REVIEW

File: src/eval/completion_tests.py
Comment (sej2020) [inline src/eval/completion_tests.py:1]: just a file used for testing

File: src/utils/translate.py
Comment (sej2020) [inline src/utils/translate.py:1]: reject this change if you would rather provide the chess boards to the stockfish eval metric another way. I'm mentioning this on my review of your PR, but I'm not sure which PR should be merged first

Comment (Xmaster6y) [reply src/utils/translate.py:1]: We'll merge the other one and resolve the conflicts.

File: src/eval/metric_completion.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import jsonlines
+import torch
+
+import src.utils.translate as translate
+from src.models.decision_transformer import DecisionTransformerConfig, DecisionTransformerModel
+
+torch.set_printoptions(sci_mode=False)
+
+
+class CompletionTest:
```
Comment (Xmaster6y) [inline src/eval/metric_completion.py:21]: Unsure of the naming here

File: src/eval/metric_completion.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            black_returns = torch.sub(black_result, black_evals)
+
+        else:
+            black_returns = self.discount ** torch.arange(black_seq_len, device=self.device) * end_rewards[1]
+            white_returns = self.discount ** torch.arange(white_seq_len, device=self.device) * end_rewards[0]
+
+        condition = torch.arange(seq_len, device=self.device) % 2 == 0
+        target_returns = torch.zeros(1, seq_len, 1, device=self.device, dtype=torch.float32)
+        target_returns[:, condition, :] = white_returns.reshape(1, white_seq_len, 1)
+        target_returns[:, ~condition, :] = black_returns.reshape(1, black_seq_len, 1)
```
Comment (Xmaster6y) [inline src/eval/metric_completion.py:146]: Return to go are actually max_return - returns

**Review by Xmaster6y — CHANGES_REQUESTED** [review]

File: src/eval/metric_completion.py (hunk 1 above)
Comment (sej2020) [reply src/eval/metric_completion.py:23]: Yeah, I'll swap it out

File: src/eval/metric_completion.py (hunk 2 above)
Comment (sej2020) [reply src/eval/metric_completion.py]: i think this is resolved now because all of the work with return to go are now in the dataset/translate module. Let me know if you think what I did in those files with return to go is correct

**Comment by sej2020** [comment]
the metric integration looks good to me. Ready for your review @Xmaster6y

**Review by Xmaster6y — COMMENTED** [review]
You need to install pre-commit so that you make commits that pass the CI

**Review by Xmaster6y — APPROVED** [review]
I'll change the modelling afterwards since I forgot to include the casting encoding.

**Review by Xmaster6y — APPROVED** [review]
I'll merge like this but feel free to open an issue
